File: barricade-annotations/src/main/java/com/mutualmobile/barricade/annotation/Barricade.java
package com.mutualmobile.barricade.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;
import java.util.HashMap;

import static java.lang.annotation.RetentionPolicy.CLASS;

/**
 * Annotation to configure Barricade for an endpoint.
 */
@Documented @Retention(CLASS) @Target(ElementType.METHOD) public @interface Barricade {
  String endpoint() default "";
  UrlPath[] path() default {};
  QueryParams[] queryParams() default {};
  RequestJson[] requestJson() default {};
  Response[] responses() default {};
}

File: barricade-annotations/src/main/java/com/mutualmobile/barricade/annotation/Barricade.kt
package com.mutualmobile.barricade.annotation

/**
 * Annotation to configure Barricade for an endpoint.
 */
@Retention
@Target(AnnotationTarget.FUNCTION)
annotation class Barricade(
    val endpoint: String = "",
    val responses: Array<Response> = []
)
